src/Map.js

```diff
@@ -368,30 +407,38 @@ export default function Map (props) {
       state.subscribtionCeremony !== state.lastCeremony.subscribtionCeremony &&
         dispatch({ type: 'reset' }); // reset state on start of new ceremony
       getCurrentCeremonyIndex().then((currentCeremonyIndex) => {
-        debug && console.log('set ceremonyIndex', currentCeremonyIndex.toString());
+        // debug && console.log('set ceremonyIndex', currentCeremonyIndex.toString());
         setCeremonyIndex(currentCeremonyIndex);
       });
     }
   }, [currentPhase.phase, ceremonyIndex]);
 
+  // OPTIONAL FOR DEBUGGING, FINDING OUT HOW THE PARTICULAR PROPERTIES OF THE STATE CHANGE
```

Comment:
This is good; leave it here. It will be helpful. In general, I prefer to use function strings, which is less verbose than using `+` to concatenate strings, e.g.:

```
console.log(`state participants are: ${JSON.stringify(state.participants))}`, [state]);
```

Why do you use `[state]` separately after the comma? `console.log` has the args: `path`, `msg`. Hence you would use it like:

`console.log('function/class where the logging takes place', 'log msg')`

Reply:
I'm not sure you know how react works here:
We are using states here (not regular variables), which are initialized with useState(state,setterOfState) =... 
You can not simply log states in react after you set them, since setting them occurs in a different time during rendering. 
The only way i found to securely log them, is with using the useEffect function. 
The function looks like this:
UseEffect( () => {},[stateOfChoise1, stateOfChoise2.. ] )
Every time at least one of the states defined in the array change, it will trigger the useEffect to excecute
UseEffect

Reply:
Ah sorry, I misread the brackets, and after your explanation it makes sense to me. Thanks!